src/java/src/main/java/ai/onnxruntime/genai/Adapters.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+package ai.onnxruntime.genai;
+
+
+public final class Adapters implements AutoCloseable {
+  private long nativeHandle = 0;
+
+  /**
+   * Constructs an Adapters object with the given model.
+   *
+   * @param model The model.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public Adapters(Model model) throws GenAIException {
+    if (model.nativeHandle() == 0) {
+      throw new IllegalArgumentException("model has been freed and is invalid");
+    }
+
+    nativeHandle = createAdapters(model.nativeHandle());
+  }
+
+  /**
+   * Load an adapter from the specified path.
+   *
+   * @param adapterFilePath The path of the adapter.
+   * @param adapterName A unique user supplied adapter identifier.
+   * @throws GenAIException If the call to the GenAI native API fails.
+   */
+  public void loadAdapters(String adapterFilePath, String adapterName) throws GenAIException {
```

Comment:
Is there an API that allows passing in bytes for the adapter so we can read from the app's assets and not need to copy to the app's filesDir so it has a valid path?

Reply:
I'll have to ask @baijumeswani do we have that? I don't think so. Reading from bytes seems a reasonable option.